qiime2/sdk/usage.py

```diff
@@ -0,0 +1,349 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import abc
+import os
+import re
+import types
+import tempfile
+import uuid
+import zipfile
+
+from qiime2 import sdk
+
+# TODO: docstrings
+
+
+class UsageAction:
+    def __init__(self, plugin_id: str, action_name: str):
+        if plugin_id == '':
+            raise ValueError('Must specify a value for plugin_id.')
+
+        if action_name == '':
+            raise ValueError('Must specify a value for action_name.')
+
+        self.plugin_id = plugin_id
+        self.action_name = action_name
+        self._plugin_manager = sdk.PluginManager()
+
+    def __repr__(self):
+        return 'UsageAction(plugin_id=%r, action_name=%r)' %\
+            (self.plugin_id, self.action_name)
+
+    def get_action(self):
+        plugin = self._plugin_manager.get_plugin(id=self.plugin_id)
+        # TODO: should this validation be pushed up into
+        # plugin.py or action.py?
```

Comment:
That's a good question, we have overridden those classes, so it would be pretty easy to super `__getitem__` with a nicer KeyError. I would be in favor I think (as long as it stays `KeyError` we won't have broken backwards compat, I think).

Reply:
seems best to defer this for now.